python_dwd/data_storing.py

```diff
@@ -0,0 +1,120 @@
+        parameter: the parameter enumeration
+        time_resolution: the time resolution enumeration
+        period_type: the period type as enumeration
+        folder: the folder where the hdf is stored
+    Returns:
+        None, prints information if data was not stored
+    """
+    # Make sure that there is data that can be stored
+    if station_data.empty:
+        print("There is no data in 'station_data' that can be stored.")
+
+        return
+
+    request_string = _build_local_store_key(
+        station_id, parameter, time_resolution, period_type)
+
+    local_filepath = Path(folder, STATIONDATA_NAME).absolute() / \
+        f"{STATIONDATA_NAME}{H5_FORMAT}"
+    local_filepath.parent.mkdir(parents=True, exist_ok=True)
+
+    try:
+        station_data.to_hdf(
+            path_or_buf=local_filepath,
+            key=request_string
+        )
+    except FileNotFoundError:
+        print(f"Error: File for station data could not be created at {str(local_filepath)}. "
+              f"Data for {request_string} could not be written.")
+
+
+def restore_dwd_data(station_id: int,
+                     parameter: Parameter,
+                     time_resolution: TimeResolution,
+                     period_type: PeriodType,
+                     folder: str) -> Tuple[bool, pd.DataFrame]:
+    """
+    Function to restore data from a local hdf file based on the place (folder) where
+    the file is stored and parameters that define the request in particular.
+
+    Args:
+        station_id: the station id of which data should be restored
+        parameter: parameter as enumeration
+        time_resolution: time resolution as enumeration
+        period_type: period type as enumeration
+        folder: folder where the hdf file should be found as string
+
+    Returns:
+        Tuple, first is boolean if data could be restored, second is the DataFrame
+        holding the data
+    """
+    request_string = _build_local_store_key(station_id, parameter, time_resolution, period_type)
+
+    local_filepath = Path(folder, STATIONDATA_NAME).absolute() / \
+        f"{STATIONDATA_NAME}{H5_FORMAT}"
+
+    try:
+        # typing required as pandas.read_hdf returns an object by typing
+        station_data: Union[object, pd.DataFrame] = pd.read_hdf(
+            path_or_buf=local_filepath,
+            key=request_string
+        )
+    except FileNotFoundError:
+        print(f"Error: There seems to be no file at {str(local_filepath)}. "
+              f"Data will be loaded freshly.")
+        return False, pd.DataFrame()
+    except KeyError:
+        print(f"Error: The requested data for {request_string} does not yet "
+              f"exist in local store at {str(local_filepath)}. "
+              f"Data will be loaded freshly.")
+        return False, pd.DataFrame()
+
+    return True, station_data.astype(create_stationdata_dtype_mapping(station_data.columns))
```

Comment:
Do we know what this boolean is for in the later stages? Would be great to understand the reason why here. Otherwise manage boolean obtaining where the empty dataFrame arrives

Reply:
The boolean is to define the success of the restore (it can only be positive or negative) and so take the analysis step from the collect_dwd_function to this function itself, so that it works somehow independent.

Reply:
I would pass here silently , also If it fails to Store data locally. When the user wants to read from local store then you can throw an info and say that Data will be downloaded bc they are not there.

Reply:
So I thought about it and I think the print statement should stay but should be shortened. If there's neither a raise nor a print, there's really no feedback. Might be that the user points to a wrong directory, which is why the library has to reload all the data...

Maybe the print could be replaced by a logging and the user could manually set it to hide for themselves.